Gems/ROS2/Code/Source/Clock/SimulationClock.cpp

```diff
@@ -49,5 +54,15 @@ namespace ROS2
         rosgraph_msgs::msg::Clock msg;
         msg.clock = GetROSTimestamp();
         m_clockPublisher->publish(msg);
+
+        uint64_t deltaTime = elapsed - m_lastExecutionTime;
+        m_lastExecutionTime = elapsed;
+
+        // statistics on execution time
+        m_frameTimes.push_back(deltaTime);
+        AZStd::set<uint64_t> frameTimeSorted{ m_frameTimes.begin(), m_frameTimes.end() };
```

Comment:
question: Rather than creating a `set` here would it not be faster to just sort the vector (or create a temp vector and sort that?). Might not be critical but just a suggestion, would be interesting to profile quickly to see what the difference is potentially 😛

Reply:
I am curious too, run little benchmark in code:
```
    {
        AZ::Debug::Timer t;
        t.Stamp();
        for (int i = 0; i < 1000; i++)
        {
            AZStd::set<AZ::s64> frameTimeSorted{ m_frameTimes.begin(), m_frameTimes.end() };
            AZStd::set<AZ::s64>::iterator it = frameTimeSorted.begin();
            std::advance(it, frameTimeSorted.size() / 2);
            m_currentMedian = *it;
        }
        AZ_Printf("SimulationClock", "set time : %f", t.GetDeltaTimeInSeconds());
    }

    {
        AZ::Debug::Timer t;
        t.Stamp();
        for (int i = 0; i < 1000; i++)
        {
            AZStd::vector<AZ::s64> frameTimeSorted{ m_frameTimes.begin(), m_frameTimes.end() };
            AZStd::sort(frameTimeSorted.begin(),frameTimeSorted.end());
            AZStd::vector<AZ::s64>::iterator it = frameTimeSorted.begin();
            std::advance(it, frameTimeSorted.size() / 2);
            m_currentMedian = *it;
        }
        AZ_Printf("SimulationClock", "vector time : %f", t.GetDeltaTimeInSeconds());
    }
``` 
Implementation with vector is 2x faster :thinking: I will add it here.

Reply:
Haha no way! 😅 That's really cool, thanks for checking 🙂